src/Compiler/Facilities/BuildGraph.fs

```diff
@@ -19,8 +20,16 @@ let node = async
 [<AbstractClass; Sealed>]
 type NodeCode =
 
+    static member RunImmediate(computation: NodeCode<'T>, ct: CancellationToken) =
+        try
+            let work = async { return! computation |> Async.AwaitNodeCode }
+
+            Async.StartImmediateAsTask(work, cancellationToken = ct).Result
+        with :? AggregateException as ex when ex.InnerExceptions.Count = 1 ->
+            raise (ex.InnerExceptions[0])
```

Comment:
We should instead use something like `ExceptionDispatchInfo.Capture(ex.InnerExceptions[0]).Throw()` to preserve original stack trace.

Reply:
Yes, this is just the original code restored. Something blocks in CI, so I tried to restore everything related to BuildGraphTests. We can iterate when this is hopefully green.